tests/cross/src/test/scala/tests/pc/AutoImplementMethodsSuite.scala

```diff
@@ -0,0 +1,506 @@
+package tests.pc
+
+import org.eclipse.{lsp4j => l}
+
+import tests.BaseCodeActionSuite
+import scala.meta.internal.metals.CompilerOffsetParams
+import scala.meta.internal.jdk.CollectionConverters._
+import scala.meta.internal.metals.TextEdits
+
+class AutoImplementMethodsSuite extends BaseCodeActionSuite {
+
+  checkEdit(
+    "classdef",
+    """|package a
+       |
+       |object A {
+       |  trait Base {
```

Comment:
Do we deal with generic classes like:
```
trait A[T]{
  def method : T
}

class B extends A[Int]{}
```

Reply:
Yes:smile:

![generics](https://user-images.githubusercontent.com/9353584/74080420-4660a280-4a87-11ea-9d51-15266f80926e.gif)

- Syntactically, `class B extends A[Int] {}` can be captured by this case: https://github.com/scalameta/metals/blob/2969ba9e3f907ff769469cd020720a6145de35ea/mtags/src/main/scala/scala/meta/internal/pc/OverrideCompletions.scala#L303-L313
- After that, the methods to implement are calculated based on semantics.

So, yes, we deal with that case.

Reply:
Added test in https://github.com/scalameta/metals/pull/1379/commits/0bd0092ecb6890e62d2b137b085fed55d4db5473